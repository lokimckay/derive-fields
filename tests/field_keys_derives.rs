mod common;

use common::*;
use derive_fields::FieldKeys;

#[derive(FieldKeys)]
#[field_keys_derives(Debug)]
pub struct ExampleStruct {
    _name: String,
    _category: Category,
    _really_really_long_key: bool,
}

fn main() {
    // Confirm that key no longer implements clone
    let key = ExampleStructFieldKey::Name;
    let _ = key.clone();
}
