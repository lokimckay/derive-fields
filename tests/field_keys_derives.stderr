error[E0599]: no method named `clone` found for enum `ExampleStructFieldKey` in the current scope
  --> tests/field_keys_derives.rs:17:17
   |
 6 | #[derive(FieldKeys)]
   |          --------- method `clone` not found for this enum
...
17 |     let _ = key.clone();
   |                 ^^^^^ method not found in `ExampleStructFieldKey`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `clone`, perhaps you need to implement it:
           candidate #1: `Clone`
